vim-patch: update runtime files

clason opened the pull request:
- **vim-patch:c18a9d5: runtime(netrw): using inefficient highlight pattern for 'mf'**
- **vim-patch:41c7bba: runtime(zsh,sh): set and unset compiler in ftplugin**
- **vim-patch:2307945: runtime(java): Optionally recognise all primitive constants in _switch-case_ labels**

clason committed:
vim-patch:c18a9d5: runtime(netrw): using inefficient highlight pattern for 'mf'

Fixes E872 too many '(' in highlight pattern for `mf` selection

fixup for vim/vim#15551
closes: vim/vim#15700

https://github.com/vim/vim/commit/c18a9d5835456e0e47e943b673d631caaebdbea3

Co-authored-by: yasuda <yasuda@kyoto-sr.co.jp>

clason committed:
vim-patch:41c7bba: runtime(zsh,sh): set and unset compiler in ftplugin

closes: vim/vim#15699

https://github.com/vim/vim/commit/41c7bbaf8f567d3a19ab2a6191ec9937fb890220

Co-authored-by: Konfekt <Konfekt@users.noreply.github.com>

clason committed:
vim-patch:2307945: runtime(java): Optionally recognise all primitive constants in _switch-case_ labels

Define "g:java_syntax_previews" and include number 455 in
its list to enable this recognition:
------------------------------------------------------------
	let g:java_syntax_previews = [455]
------------------------------------------------------------

Reference:
https://openjdk.org/jeps/455

closes: vim/vim#15698

https://github.com/vim/vim/commit/23079450a8dea98f9d4e3e74ab57866914b0a813

Co-authored-by: Aliaksei Budavei <0x000c70@gmail.com>

clason committed:
vim-patch:f86568f: runtime(misc): simplify keywordprg in various ftplugins

closes: vim/vim#15696

https://github.com/vim/vim/commit/f86568f91848ece0c5da2178881b3ed858dae799

Co-authored-by: Konfekt <Konfekt@users.noreply.github.com>

zeertzjq commented:
There is a syntax error in the last commit:
```
line   26:
E580: :endif without :if: endif
```
I'd suggest you move it to a separate PR.

clason committed:
vim-patch:87b6565: runtime(modconf): remove erroneous :endif in ftplugin

regression introduced in f86568f91848ece0c5da

closes: vim/vim#15704

https://github.com/vim/vim/commit/87b65652e806ed5ae999f8b1f46b9306d48c9326

Co-authored-by: zeertzjq <zeertzjq@outlook.com>

clason committed:
vim-patch:597aadc: runtime(lyrics): support multiple timestamps in syntax script

Problem:  Multiple timestamps in the same line were not highlighted
Solution: Adapt the syntax to support multiple timestamps

fixes: vim/vim#15703
closes: vim/vim#15707

https://github.com/vim/vim/commit/597aadcf214cd13bf6a2155bf45173edc03199de

Co-authored-by: ObserverOfTime <chronobserver@disroot.org>

clason committed:
vim-patch:50423ab: runtime(java): Optionally recognise _module_ import declarations

Define "g:java_syntax_previews" and include number 476 in
its list to enable this recognition:
------------------------------------------------------------
        let g:java_syntax_previews = [476]
------------------------------------------------------------

Reference:
https://openjdk.org/jeps/476

closes: vim/vim#15709

https://github.com/vim/vim/commit/50423ab8086c6e51b86548de77d79db8542ae317

Co-authored-by: Aliaksei Budavei <0x000c70@gmail.com>

clason commented:
Added your fix here.
